### PR TITLE
Perf(dflash): compile create_block_mask to avoid dense mask materialization

### PR DESCRIPTION
**Pull request by WindChimeRan**
## Purpose

DFlash/DSpark rebuild the attention mask every training step. Uncompiled, `create_block_mask` evaluates the mask over the full Q × KV grid. For`--speculator-type dspark` on Qwen3-8B (`--max-anchors 3072 --block-size 8
--total-seq-len 4096`):
```
    Q  = max_anchors × block_size  = 3072 × 8       = 24576
    KV = total_seq_len + Q         = 4096 + 24576   = 28672
```
So a 24576 × 28672 dense grid (~7 GB) is built and thrown away every step.

`torch.compile` fuses the mask evaluation with the block-sparsity reduction, so the BlockMask is produced without ever materializing that grid: peak mask memory drops from ~7 GB to ~0, and the mask is bit-identical.

**Not applicable to EAGLE3/PEAGLE** — their autoregressive drafting builds only a sequence-sized mask, so there's nothing to fix (benchmarked: no gain).

## Measured (DSpark, Qwen3-8B, 5-layer draft, seq 4096, 2×A100)

| metric      | before (dense)   | after (compiled) |
| ----------- | ---------------- | ---------------- |
| mask build  | 29.8 ms / 7.05 GB | 2.9 ms / ~0 GB   |
| forward     | 357.9 ms         | 317.3 ms (−11%)  |
| step        | 1087.5 ms        | 1047.9 ms (−3.6%) |

## Tests

just `torch.compile` a function `create_block_mask`. No behavior change. 

## Reference

https://pytorch.org/blog/flexattention/

(the blog is stale, now create_block_mask is compilable)

## Checklist

I have filled in:

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan/results, such as providing test command and pasting the results.
- [ ] (Optional) The necessary documentation update.
- [x] I (a human) have written or reviewed the code in this pr to the best of my ability.

### CODE REVIEW

File: src/speculators/models/dflash/core.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
 from speculators.models.metrics import LossConfig, resolve_loss_config
 from speculators.models.utils import conditional_torch_compile, resolve_target_layer_ids
 
+# Compile so the mask builds block-sparse instead of materializing DFlash's huge
+# dense [Q, KV] grid every step. (No benefit for EAGLE3's small autoregressive mask.)
+_compiled_create_block_mask = torch.compile(create_block_mask)
+
```
[the remark this answers is not in the document]
Comment (WindChimeRan) [reply src/speculators/models/dflash/core.py:31]: shapes are fixed: q_len=num_anchors*block_size, kv_len=total_seq_len+q_len

**Review by fynnsu — APPROVED** [review]
How did you test this?

**Comment by WindChimeRan** [comment]
```python
# repro_mask.py  —  run:  python repro_mask.py
import torch
from torch.nn.attention.flex_attention import create_block_mask
from speculators.models.dflash.attention import create_anchor_block_mask_mod

device = "cuda"
# DSpark / Qwen3-8B config from the PR: --max-anchors 3072 --block-size 8 --total-seq-len 4096
max_anchors, block_size, total_seq_len = 3072, 8, 4096

document_ids = torch.zeros(total_seq_len, dtype=torch.long, device=device)
anchor_positions = torch.linspace(0, total_seq_len - 1, max_anchors, device=device).long()

mask_mod, q_len, kv_len = create_anchor_block_mask_mod(
    document_ids=document_ids, total_seq_len=total_seq_len,
    anchor_positions=anchor_positions, block_size=block_size,
)
print(f"Q_LEN={q_len}  KV_LEN={kv_len}")

def build(fn, reps=5):
    fn(mask_mod, B=None, H=None, Q_LEN=q_len, KV_LEN=kv_len, device=device)  # warm up
    torch.cuda.synchronize(); torch.cuda.reset_peak_memory_stats()
    best = float("inf")
    for _ in range(reps):
        s, e = torch.cuda.Event(enable_timing=True), torch.cuda.Event(enable_timing=True)
        s.record(); bm = fn(mask_mod, B=None, H=None, Q_LEN=q_len, KV_LEN=kv_len, device=device); e.record()
        torch.cuda.synchronize(); best = min(best, s.elapsed_time(e))
    return bm, best, torch.cuda.max_memory_allocated() / 1e9

bm_eager, t_eager, m_eager = build(create_block_mask)               # dense grid materialized
bm_comp,  t_comp,  m_comp  = build(torch.compile(create_block_mask)) # fused, grid never built

print(f"eager    : {t_eager:8.1f} ms   peak {m_eager:6.3f} GB")
print(f"compiled : {t_comp:8.1f} ms   peak {m_comp:6.3f} GB")
print(f"mem saved: {m_eager - m_comp:.3f} GB    speedup {t_eager / t_comp:.0f}x (isolated build)")
print("masks bit-identical:", all(torch.equal(a, b)
      for a, b in zip(bm_eager.as_tuple(), bm_comp.as_tuple()) if isinstance(a, torch.Tensor)))
```

```
Q_LEN=24576  KV_LEN=28672
eager    :    145.8 ms   peak  7.048 GB
compiled :      2.4 ms   peak  0.003 GB
mem saved: 7.045 GB    speedup 61x (isolated build)
masks bit-identical: True
```
